Update sportssync extension

daniyalmaster693 opened the pull request:
## Description

<!-- A summary of your change. If you add a new extension or command, explain what it does. -->

## Screencast

<!-- If you add a new extension or command, include a screencast (or screenshot for straightforward changes). A good screencast will make the review much faster - especially if your extension requires registration in other services.  -->

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

daniyalmaster693 commented:
I have question: I refactored and rewrote a lot of code since the last update, so the setup of commands is a lot more modular. However, the issue I've had is I'm not sure how to handle the isLoading attribute. Right now its a mess, and most commands don't have it.

daniyalmaster693 commented:
Sorry it's been a while, had some exams recently, so haven't had time to work on the extension. But I've gone through and applied the changes greptile suggested. Could you take a look?

pernielsentikaer reviewed (changes requested):
How should the menubar work? I didn't select a sport and then I'll see this

![CleanShot 2025-04-11 at 08 44 00@2x](https://github.com/user-attachments/assets/386276d6-1d44-4d7e-a178-cb56886a8c9a)

Which calls this https://site.api.espn.com/apis/site/v2/sports/default/default/scoreboard

daniyalmaster693 commented:
https://github.com/user-attachments/assets/1a2fb9c0-a28a-4639-8f71-40832c366db3

daniyalmaster693 commented:
Yes it only displays one league, which is the one you select from the preferences

pernielsentikaer commented:
Is this used anywhere else? It should not make any network calls if you select none

daniyalmaster693 commented:
The preferences are also used for the favorite team dashboard command. The reason it makes network calls is that it's just inputting "none" into the api call

daniyalmaster693 commented:
I removed the none option from the preferences settings, and I replaced it with the first option for each dropdown (so the default is now: hockey, nhl, anaheim ducks). That way both the live scores and favorite team dashboard commands will still work

pernielsentikaer commented:
It still feels a bit cluttered, why does it show Select a game in the menu bar, there is also only one possibility to select - I'm a bit confused actually 😊

![CleanShot 2025-04-14 at 08 08 16@2x](https://github.com/user-attachments/assets/902db2e9-c810-40ef-a986-7636a14e6bf9)

daniyalmaster693 commented:
The games are within the submenu (hover over it) amd you'll see them. From there you can select a game

pernielsentikaer commented:
When you haven't selected a game, then we shouldn't make any fetch calls, should we? I can see that it calls `https://site.api.espn.com/apis/site/v2/sports/hockey/nhl/scoreboard` pretty often

daniyalmaster693 commented:
I agree, but I'm not sure how to achieve that, since it's fetching the api call each time the menubar refreshes, which is updating the game item's scores.

pernielsentikaer commented on extensions/sportssync/src/live-scores-menubar.tsx:
```diff
@@ -0,0 +1,321 @@
+
+// Get User Preferences
+
+interface preferences {
+  name: string;
+  id?: string;
+}
```
You don't need to manually set preference types as this is autogenerated in `raycast-env.d.ts`

```suggestion
```

pernielsentikaer commented on extensions/sportssync/src/live-scores-menubar.tsx:
```diff
@@ -0,0 +1,321 @@
+    isLoading: scheduleLoading,
+    data: scheduleData,
+    revalidate: scheduleRevalidate,
+  } = useFetch<Response>(`https://site.api.espn.com/apis/site/v2/sports/${favoriteSport}/${favoriteLeague}/scoreboard`);
```
there is an execute param you can use to check if it should run or not

daniyalmaster693 commented:
But this isn't needed since I removed the option to select none all together. A default league, team and sport is already assigned (hockey, nhl, Anaheim ducks), therefore when it fetches game information, its returning actual info.
